Промежуточная аттестация №1. Кручинин С.В.
Разведочный анализ по базе библиотеки. Это блокнот к уже сданному разбору.
Запускать из папки аттестация_1.


In [1]:
import pandas as pd
df = pd.read_csv('priznaki.csv')
print('строк', len(df), 'колонок', df.shape[1])
print(df['prosrochka'].value_counts().sort_index())
print('пустых оценок', int(df['otsenka'].isna().sum()))
print(df.groupby('prosrochka')[['vozrast', 'otsenka', 'tema_sovpala', 'skok_brali', 'sred_ocenka_knigi']].mean().round(2))
rated = df[df['otsenka'].notna()]
print('средняя оценка, тема совпала или нет')
print(rated.groupby('tema_sovpala')['otsenka'].mean().round(2))
print('срок, дни', sorted(df['srok_dney'].unique()))
print('возраст от', int(df['vozrast'].min()), 'до', int(df['vozrast'].max()))


строк 96 колонок 22
prosrochka
0    80
1    16
Name: count, dtype: int64
пустых оценок 15
            vozrast  otsenka  tema_sovpala  skok_brali  sred_ocenka_knigi
prosrochka                                                               
0             30.04     3.93          0.20        5.69               3.83
1             20.31     2.83          0.25        5.69               3.61
средняя оценка, тема совпала или нет
tema_sovpala
0    3.51
1    4.55
Name: otsenka, dtype: float64
срок, дни [np.int64(14)]
возраст от 17 до 45


Что видно. 96 выдач: 80 без просрочки, 16 с просрочкой. Без просрочки возраст около 30 и оценка 3.93. С просрочкой возраст около 20 и оценка 2.83. Если тема совпала, оценка 4.55, если нет — 3.51. Пустых оценок 15: книга ещё на руках.
